integer_validator Function Documentation and Tests
===========================================

This document provides documentation and doctests for the `integer_validator` function.


Doctests:
---------
Here are some examples demonstrating the usage of the `integer_validator` function:
>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry

    >>> bg = BaseGeometry()
    >>> bg.integer_validator("my_int", 12)


    >>> bg = BaseGeometry()
    >>> bg.integer_validator("name", "John")
    Traceback (most recent call last):
        ...
    TypeError: name must be an integer

    >>> bg = BaseGeometry()
    >>> bg.integer_validator("age", 0)
    Traceback (most recent call last):
        ...
    ValueError: age must be greater than 0

    >>> bg = BaseGeometry()
    >>> bg.integer_validator("distance", -4)
    Traceback (most recent call last):
        ...
    ValueError: distance must be greater than 0

    >>> bg = BaseGeometry()
    >>> bg.integer_validator("distance", 4.9)
    Traceback (most recent call last):
        ...
    TypeError: distance must be an integer

    >>> bg = BaseGeometry()
    >>> bg.integer_validator("age")
    Traceback (most recent call last):
        ...
    TypeError: integer_validator() missing 1 required positional argument: 'value'

    >>> bg = BaseGeometry()
    >>> bg.integer_validator(27)
    Traceback (most recent call last):
        ...
    TypeError: integer_validator() missing 1 required positional argument: 'value'

    >>> bg = BaseGeometry()
    >>> bg.integer_validator()
    Traceback (most recent call last):
        ...
    TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'

    >>> bg = BaseGeometry()
    >>> bg.integer_validator("my_int", True)
    Traceback (most recent call last):
        ...
    TypeError: my_int must be an integer
